### PR TITLE
Return only summary of existing searches/dashboards

### LINKED ISSUE #7051: Return only summary of existing searches/dashboards.
When the views model is extended in a feature branch, switching back to another branch leads to the dashboards/search list not being retrievable from the backend, due to mapping errors. This leads to unneeded friction. We should use a simplified model for listing existing searches/dashboards, that is lenient when it comes to additional properties.

Search/dashboard endpoints should read & return only summary (id, title, description) of entities, to:  
- reduce the size of the payload for listing entities
- prevent that the addition/removal of fields in the views makes listing them fail

### LINKED ISSUE #10338: SearchesCleanupJob should use summary DTOs to prevent deserialization errors
Currently, the `SearchesCleanupJob` reads full `ViewDTO`/`Search` objects when determining the expired searches it needs to delete. This creates unnecessary load and results in potential deserialization errors when the schema/class hierarchy has changed in an incompatible way.

Thus, the `SearchesCleanupJob` should use stripped down summary DTOs.

### PR DESCRIPTION
**Pull request by janheise**
In places where only a subset and not all object's attributes is needed, only summaries of Searches and Dashboards are loaded and transferred between backend and frontend.

## Description
A new ViewSummaryDTO object is introduced that holds only a subset of the ViewDTO object's attributes. 
Also, a new ViewSummaryService i introduced and injected into the DashboardResource and SearchResource.
The requirements for View on the frontend-side have been loosened so that missing fields that had been required in the past are no longer necessary. Fixes #10338 , Fixes #7051

## Motivation and Context
Motivation is twofold:
1. when new fields are introduced to Views (e.g. feature branches) and documents with those fields exist in the db, loading those documents fail after switching to a different branch that does not hold those changes in structure.
2. the payload to transport those DTOs to the frontend is reduced as only the necessary attributes for displaying them are loaded from the db and serialized/deserialized

## How Has This Been Tested?
1. manually tested graylog after the changes that the functionality still works as expected in the GUI with valid objects
2. introduced invalid objects into MongoDB and tested again

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, works well! Just a couple of minor nitpicks left.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/db/SearchDbService.java [hunk 1]
```diff
@@ -117,4 +124,15 @@ private Search requirementsForSearch(Search search) {
         return searchRequirementsFactory.create(search)
                 .rebuildRequirements(Search::requires, (s, newRequirements) -> s.toBuilder().requires(newRequirements).build());
     }
+
+    Stream<SearchSummary> findSummaries() {
+        return Streams.stream((Iterable<SearchSummary>) summarydb.find());
+    }
+
+    public Set<String> getExpiredSearches(final Set<String> requiredIds, final Instant mustNotBeOlderThan) {
```
[the remark this answers is not in the document]
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/db/SearchDbService.java:132]: I vote for "neverDeleteIds"

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchSummaryRequirements.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package org.graylog.plugins.views.search;
+
+import com.google.inject.assistedinject.Assisted;
+import org.graylog.plugins.views.Requirement;
+import org.graylog.plugins.views.Requirements;
+
+import javax.inject.Inject;
+import java.util.Set;
+
+public class SearchSummaryRequirements extends Requirements<SearchSummary> {
```
[the remark this answers is not in the document]
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchSummaryRequirements.java:26]: You're correct, it should be obsolete now.

**Review by dennisoelkers — APPROVED** [review]
Looks good, works great! :tada:

**Pull request merged**